src/main/clojure/clara/rules/engine.cljc

```diff
@@ -118,6 +118,14 @@
   ;; Right-activate the node with items reduced in the above pre-reduce step.
   (right-activate-reduced [node join-bindings reduced  memory transport listener]))
 
+(defprotocol IAccumInspect
+  ;; Takes a token that was previously propagated from the node
```

Comment:
"Also, am I correct in understanding this accepts a token from this node, or a descendent of that token, and returns the corresponding elements?"

That is mostly correct.  It returns the matching facts, not [Element records](https://github.com/cerner/clara-rules/blob/0.14.0/src/main/clojure/clara/rules/engine.cljc#L26).

I'll try to clear this up; let me know if you have further suggestions.  I made some comments in the implementations on AccumulateNode and AccumulateWithJoinFilterNode that could be helpful in understanding this as well.